apps/studio/components/interfaces/SQLEditor/SQLEditor.tsx

```diff
@@ -94,9 +94,13 @@ export const SQLEditor = () => {
   const snapV2 = useSqlEditorV2StateSnapshot()
   const getImpersonatedRoleState = useGetImpersonatedRoleState()
   const databaseSelectorState = useDatabaseSelectorStateSnapshot()
-  const isOptedInToAI = useOrgOptedIntoAi()
+  const aiOptInLevel = useOrgAiOptInLevel()
   const [selectedSchemas] = useSchemasForAi(project?.ref!)
-  const includeSchemaMetadata = isOptedInToAI || !IS_PLATFORM
+  const includeSchemaMetadata =
+    aiOptInLevel === 'schema' ||
+    aiOptInLevel === 'schema_and_log' ||
+    aiOptInLevel === 'schema_and_log_and_data' ||
+    !IS_PLATFORM
```

Comment:
i feel like this can be returned as part of the `useOrgAiOptInLevel` hook since its a repeated logic in multiple places (SQLEditor, edge functions new.tsx + code.tsx)

DX at the top of my head: `const { aiOptInLevel, includeSchemaMetadata } = useOrgAiOptInLevel()`